instrumentation/spring-amqp/src/main/java/brave/spring/amqp/TracingRabbitListenerAdvice.java

```diff
@@ -0,0 +1,62 @@
+package brave.spring.amqp;
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext.Extractor;
+import brave.propagation.TraceContextOrSamplingFlags;
+import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
+
+/**
+ * TracingRabbitListenerAdvice is an AOP advice to be used with the {@link
+ * SimpleMessageListenerContainer#setAdviceChain(Advice...) SimpleMessageListenerContainer} in order
+ * to add tracing functionality to a spring-amqp managed message consumer. While a majority of
+ * brave's instrumentation points is implemented using a delegating wrapper approach, this extension
+ * point was ideal as it covers both programmatic and @RabbitListener annotation driven amqp consumers.

```

Comment:
`@RabbitListener` won't do what you want in javadoc. If you want it to look like that I think you need to wrap it in `{@code @RabbitListener}`